### PR TITLE
Add to_nmatrix method in Daru::Vector

### PR DESCRIPTION
**Pull request by lokeshh**
Daru::DataFrame and Daru::Vector has `to_matrix` method to convert to `Matrix` object but no method in Daru::Vector to convert to `NMatrix` object. This PR accomplishes that. In addition it adds `is_numeric?` method to decide if vector contain all numeric values or not.

### CODE REVIEW

**Comment by lokeshh** [comment]
This method will be useful to replace `Matrix` with `NMatrix` in Statsample-GLM.

**Comment by v0dro** [comment]
In Ruby you can get away with just `numeric?` instead of `is_numeric?` since it allows question marks.

Make that change and I'll merge.

**Comment by lokeshh** [comment]
I just found out that there's already `numeric?` implemented in Daru. So, I've removed `is_numeric?` completely and now using `numeric?` now.

File: lib/daru/vector.rb [hunk 1]
```diff
@@ -830,6 +830,30 @@ def to_matrix axis=:horizontal
...
+    # @example
+    #   dv = Daru::Vector.new [1, 2, 3]
+    #   dv.to_nmatrix
+    #   # =>
+    #   # [
+    #   #   [1, 2, 3] ]
+    def to_nmatrix axis=:horizontal
+      raise ArgumentError, 'Can not convert to nmatrix'\
+        'because the vector is numeric' unless numeric? && !include?(nil)
+
```
Comment (lokeshh) [inline lib/daru/vector.rb]: @v0dro `numeric?` return true if vector contain `nil`s. Should this behavior be changed?

Comment (v0dro) [inline lib/daru/vector.rb]: No let it be. `nil` represents missing data, which can be numeric. Moreover, stats method work when missing data is present and they will malfunction if `numeric?` does not work the way it does.

**Comment by v0dro** [comment]
Rebase against current master and I'll merge.

File: lib/daru/vector.rb (hunk 1 above)
Comment (lokeshh) [inline lib/daru/vector.rb]: ok

**Comment by lokeshh** [comment]
This can be merged now. Also could a Daru release be done now, so I can make sure Statsample-GLM build is passing and Formula Language PR can be merged.

**Pull request merged**